### PR TITLE
MBS-12059 / MBS-12063: Autoselect YesAsia and Target as "purchase for mail-order" + sidebar

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-12059 / MBS-12063

See commit messages for details.

### CODE REVIEW

File: lib/MusicBrainz/Server/Data/URL.pm [hunk 1]
```diff
@@ -161,6 +161,7 @@ my %URL_SPECIALIZATIONS = (
     'SpiritOfMetal'       => qr{^https?://(?:www\.)?spirit-of-metal\.com/}i,
     'SpiritOfRock'        => qr{^https?://(?:www\.)?spirit-of-rock\.com/}i,
     'Stage48'             => qr{^https?://(?:www\.)?stage48\.net/}i,
+    'Target'              => qr{^https?://(?:www\.)?target\.com/}i,
```
Comment (yvanzo) [inline lib/MusicBrainz/Server/Data/URL.pm:164]:
```suggestion
    'Target'              => qr{^https?://(?:(?:intl|www)\.)?target\.com/}i,
```

File: root/static/scripts/edit/URLCleanup.js [hunk 2]
```diff
@@ -3966,6 +3966,38 @@ const CLEANUPS: CleanupEntries = {
       };
     },
   },
+  'target': {
+    match: [new RegExp(
+      '^(https?://)?(www\\.)?target\\.com/(b|p)',
```
Comment (yvanzo) [inline root/static/scripts/edit/URLCleanup.js:3971]:
```suggestion
      '^(https?://)?((intl|www)\\.)?target\\.com/(b|p)',
```

File: root/static/scripts/edit/URLCleanup.js [hunk 3]
```diff
@@ -3966,6 +3966,38 @@ const CLEANUPS: CleanupEntries = {
...
+      'i',
+    )],
+    restrict: [LINK_TYPES.mailorder],
+    clean: function (url) {
+      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?target\.com\//, 'https://www.target.com/');
```
Comment (yvanzo) [inline root/static/scripts/edit/URLCleanup.js:3976]:
```suggestion
      url = url.replace(/^(?:https?:\/\/)?(?:intl\.|www\.)?target\.com\//, 'https://www.target.com/');
```

**Review by yvanzo — COMMENTED** [review]
* There currently are 25 [“intl.target.com” links in the database](http://musicbrainz.org/ws/2/url?query=url:*intl.target.com*&fmt=json). Made code suggestions to handle them, but then could not find at least one valid example.

* The commit message about yesasia doesn’t say much to me.

**Comment by reosarevok** [comment]
Updated for both issues - I found one working intl. link and it redirected, so implemented it as a cleanup to www. as you suggested.

**Review by yvanzo — APPROVED** [review]
Got it, thanks!

:ship:

**Pull request merged**